### PR TITLE
Merge release 3.5 to release 3.6

### PR DESCRIPTION
**Pull request by sagarkapare**
Conflicts in classes 

```
ProgramLifeCycleHttpHandler:getStatus()
DistributedFlowProgramRunner
ProgramRuntimeService
DefaultSchedulerService
ScheduleTaskRunner
StreamSizeScheduler
ApplicationLifeCycleService
AppLifeCycleServiceTest
HBaseQueueDebugger scanAllQueues method
```

Mostly the conflicts were around the ID classes. The classes from release 3.5 were using getEntityId method.

### CODE REVIEW

**Comment by sagarkapare** [comment]
Build is running here - https://builds.cask.co/browse/CDAP-RUT238-1

**Comment by sagarkapare** [comment]
Build is green.

**Comment by rohitsinha54** [comment]
My changes look good.

**Comment by anwar6953** [comment]
LGTM

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/ProgramLifecycleService.java [hunk 1]
```diff
@@ -567,6 +572,90 @@ public void setInstances(ProgramId programId, int instances, @Nullable String co
...
+
+    ScheduleSpecification scheduleSpec = appSpec.getSchedules().get(scheduleName);
+    if (scheduleSpec == null) {
+      throw new NotFoundException(scheduleName, String.format("Schedule: %s for application: %s",
+                                                              scheduleName, applicationId));
+    }
+
+    String programName = scheduleSpec.getProgram().getProgramName();
+    ProgramType programType = ProgramType.valueOfSchedulableType(scheduleSpec.getProgram().getProgramType());
+    ProgramId programId = new ProgramId(namespaceId, appId, programType, programName);
```
Comment (ecfm) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/ProgramLifecycleService.java:599]: should use `ProgramId programId = applicationId.program(programType, programName);`

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/ProgramLifecycleService.java [hunk 2]
```diff
@@ -567,6 +572,90 @@ public void setInstances(ProgramId programId, int instances, @Nullable String co
...
+    }
+
+    ScheduleSpecification scheduleSpec = appSpec.getSchedules().get(scheduleName);
+    if (scheduleSpec == null) {
+      throw new NotFoundException(new ScheduleId(namespaceId, appId, scheduleName));
+    }
+
+    String programName = scheduleSpec.getProgram().getProgramName();
+    ProgramType programType = ProgramType.valueOfSchedulableType(scheduleSpec.getProgram().getProgramType());
+    ProgramId programId = new ProgramId(namespaceId, appId, programType, programName);
```
Comment (ecfm) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/ProgramLifecycleService.java:631]: ditto

Comment (ecfm) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/ProgramLifecycleService.java:631]: should create an ApplicationId object and reuse it here as mentioned above

**Pull request merged**